invenio_rdm_records/records/systemfields/record_deletion_status.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""System field for managing the record deletion status."""
+
+import enum
+
+from invenio_records.systemfields import SystemField
+
+
+class RecordDeletionStatusEnum(enum.Enum):
+    """Enumeration of a record's possible deletion states."""
+
+    PUBLISHED = "P"
+    DELETED = "D"
+    MARKED = "X"
```

Comment:
The names here are taken from the [RFC](https://codimd.web.cern.ch/y9vYxSjPQ-ytxTZOzzinhA#Data-model).
I've taken the liberty of shortening `MARKED_FOR_PURGE` to `MARKED`, because the former seems a bit unwieldy.
However, just `MARKED` may be ambiguous, and I'm open for suggestions!

Reply:
I'm fine with that.....like the X :-) X for marked